UnitTests/BraintreePayPalTests/BTPayPalClient_Tests.swift

```diff
@@ -638,6 +638,19 @@ class BTPayPalClient_Tests: XCTestCase {
         }
     }
 
+    func testRequestType_Checkout() {
```

Comment:
Ideally we'd want to test the behavior, not the implementation of BTPayPalClient. So the unit tests shouldn't care _how_ BTPayPalClient passes the vault value to the APIClient.sendAnalyticsEvent, just that it does.

Psuedocode example would be something like:
```
func testTokenize_whenVaultRequest_setsVaultAnalyticsTag() {
    let vaultRequest = BTPayPalVaultRequest()
    let _ = try? await sut.tokenize(vaultRequest)

    XCTAssertTrue(mockApiClient.lastPostedVaultType)
}

func testTokenize_whenCheckoutRequest_setsVaultAnalyticsTag() {
    let checkoutRequest = BTPayPalCheckoutRequest()
    let _ = try? await sut.tokenize(checkoutRequest)

    XCTAssertFalse(mockApiClient.lastPostedVaultType)
}
```

This will require some modification to our [MockAPIClient here](https://github.com/braintree/braintree_ios/blob/c2a675c8757f7a8b949a5853617fb55a3edec4c7/UnitTests/BraintreeTestShared/MockAPIClient.swift#L85-L95).

Also, instead of adding 2 new functions - we _could_ also just tack on the assertion to an existing test (since there are already some that use a BTPayPalCheckoutRequest & BTPayPalVaultRequest.

Reply:
This is not a huge deal for this PR, but I thought it might be useful to explain. We want to test the behavior so that if we ever need to change the internal logic/implementation, our tests still give us value.

Reply:
Thanks for the detailed explanation! I wrote a test but Xcode is stuck in a loop trying when I test, but here it is
```
    func testTokenize_whenVaultRequest_setsVaultAnalyticsTag() async {
        let vaultRequest = BTPayPalVaultRequest()
        let _ = try? await payPalClient.tokenize(vaultRequest)

        XCTAssertTrue(mockAPIClient.lastPostedVaultType)
    }
```
Will do the same for Checkout and Venmo request